yjit/src/codegen.rs

```diff
@@ -3786,51 +3782,55 @@ fn gen_send_iseq(jit: &mut JITState, ctx: &mut Context, cb: &mut CodeBlock, ocb:
 
         if unsafe { vm_ci_flag(ci) } & VM_CALL_KWARG != 0 {
             // Check that the size of non-keyword arguments matches
-            if lead_num != argc - kw_arg->keyword_len {
+            if lead_num != argc - unsafe { get_cikw_keyword_len(kw_arg) } {
                 gen_counter_incr!(cb, send_iseq_complex_callee);
                 return CantCompile;
             }
 
             // This is the list of keyword arguments that the callee specified
             // in its initial declaration.
-            const ID *callee_kwargs = keyword->table;
+            let callee_kwargs = unsafe { (*keyword).table };
 
             // Here we're going to build up a list of the IDs that correspond to
             // the caller-specified keyword arguments. If they're not in the
             // same order as the order specified in the callee declaration, then
             // we're going to need to generate some code to swap values around
             // on the stack.
-            ID *caller_kwargs = ALLOCA_N(VALUE, kw_arg->keyword_len);
-            for (int kwarg_idx = 0; kwarg_idx < kw_arg->keyword_len; kwarg_idx++)
-                caller_kwargs[kwarg_idx] = SYM2ID(kw_arg->keywords[kwarg_idx]);
+            let kw_arg_keyword_len:usize = unsafe { get_cikw_keyword_len(kw_arg) }.try_into().unwrap();
+            let mut caller_kwargs: Vec<ID> = vec![0; kw_arg_keyword_len.try_into().unwrap()];
```

Comment:
```suggestion
            let kw_arg_keyword_len:usize = unsafe { get_cikw_keyword_len(kw_arg) }.try_into().unwrap();
            let mut caller_kwargs: Vec<ID> = vec![0; kw_arg_keyword_len];
```
Should already be `usize`